### ISSUE TITLE
Chat modal UI test is unreliable 

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#3346]
### Share your bug report, feature request, or comment.

The chat modal UI test is not always rendered exactly the same. The scroll position can change between one run and another, causing the tests to not pass. It would be great to fix this test.

See storybook for the Chat Modal.

![image](https://github.com/owncast/owncast/assets/414923/855fcd73-9fff-407a-a603-1171669da682)

**Comment by 0scvr** [comment]
Just to clarify, is this the chat modal you're talking about https://github.com/owncast/owncast/blob/develop/web/components/modals/ChatModal/ChatModal.stories.tsx ?

**Comment by gabek** [comment]
Yeah! That's the one.

**Comment by 0scvr** [comment]
Can't seem to find were/when the Chat Modal test is run.

**Comment by gabek** [comment]
Sorry about that. It's used for mobile. A "Chat" button appears when the viewport is narrow and when pressing that button the modal will appear to allow people to chat.

**Comment by 0scvr** [comment]
> Sorry about that. It's used for mobile. A "Chat" button appears when the viewport is narrow and when pressing that button the modal will appear to allow people to chat.

What I meant was, how do i execute the tests to figure out where the problem is ? I can run storybook UI but can't see where/how to run the "tests that do not pass".

**Comment by gabek** [comment]
"Tests" in this case are a series of snapshots that compare the screenshot of a component previously to the screenshot now. See the screenshot above for an example of the previous screenshot being compared to a new one. Because the scroll position jumps around between snapshots it always says that the component changed, when it didn't. It's just the scroll position that's jumping around.

These snapshot comparisons are done by [Chromatic](https://629132c6e23893003a9e89c5-zofwvmjjhb.chromatic.com). Because Chromatic is a 3rd party service, you can't run it locally, but you can view and edit the stories locally.

**Comment by Martin-MacDonald** [comment]
Having a cursory glance it feels like this is coming from the fact that the ChatContainer component has a setTimeout that after 500ms scrolls to the bottom. If chromatic snapshots are taken at slightly different points then this could explain the difference.

**Comment by Martin-MacDonald** [comment]
I think you can add a timeout to chromatic before it takes the snapshot. Want to assign this to me and I can see if that fixes it?

**Comment by gabek** [comment]
That'd be awesome, @Martin-MacDonald. Thanks so much!

**Comment by Martin-MacDonald** [comment]
@gabek one question. Do the chromatic tests run on every PR? Assuming there is no way I can access them?

**Comment by gabek** [comment]
They're only run when the web project is updated. The workflow can be found [here](https://github.com/owncast/owncast/blob/develop/.github/workflows/chromatic.yml) and the Storybook file can be found within the directory of each component that's being tested.

**Comment by Martin-MacDonald** [comment]
@gabek do you know how often the tests were flakey? I have put up a PR with a delay of 1s which should allow the component to fully resolve before Chromatic takes the snapshot https://github.com/owncast/owncast/pull/3488 ... if they were not always flakey though might get a false positive?

**Comment by Martin-MacDonald** [comment]
Ah I was able to get it to see the snapshot but doesnt seem to have fixed it

**Comment by gabek** [comment]
It's completely random. Certainly the definition of a flakey test :)

**Comment by Martin-MacDonald** [comment]
@gabek ok have tried with setting the delay which made no difference. I ran storybook locally and it works as expected then I checked on chromatic and turns out the canvas is as expected but the snapshot is different from the canvas. Checked on GitHub and seems this had happened to other people most notably in these issues https://github.com/chromaui/chromatic-cli/issues/680 and https://github.com/chromaui/chromatic-cli/issues/275. The latter one is closed with the message to get in touch with support to query why a snapshot would differ randomly from the canvas. I obviously don't have access to the account so might be worth someone raising a ticket with chromatic to see if it is an issue their end.
